uvtools/dspec.py

```diff
@@ -1,78 +1,160 @@
+                d_mdl[i] = np.fft.fft(_d_cl)
+                del info_here['res']
+                info.append(info_here)
+    else: 
+        raise ValueError('data must be a 1D or 2D array')
     d_res = data - d_mdl * wgts
+
     return d_mdl, d_res, info
 
-def delay_filter_aa(aa, data, wgts, i, j, sdf, phs2lst=False, jds=None, 
-        skip_wgt=0.5, lst_res=binning.DEFAULT_LST_RES, standoff=0., horizon=1., 
-        tol=1e-4, window='none', maxiter=100):
-    '''Use information from AntennaArray object to delay filter data, with the
-    option to phase data to an lst bin first.  Arguments are the same as for
-    delay_filter and binning.phs2lstbin.  Returns mdl, residual, and info
-    in the frequency domain.'''
-    if phs2lst:
-        data = binning.phs2lstbin(data, aa, i, j, jds=jds, lst_res=lst_res)
-    bl = aa.get_baseline(i,j)
-    return delay_filter(data, wgts, np.linalg.norm(bl), sdf, 
-            standoff=standoff, horizon=horizon, tol=tol, window=window, 
-            skip_wgt=skip_wgt, maxiter=maxiter)
+
+    
+def delay_filter(data, wgts, bl_len, sdf, standoff=0., horizon=1., tol=1e-4, 
+        window='none', skip_wgt=0.5, maxiter=100):
+    '''Apply a wideband delay filter to data. Variable names preserved for 
+        backward compatability with capo/PAPER analysis.
+
+    Arguments:
+        data: 1D or 2D (real or complex) numpy array to where last dimension is frequency.
+        wgts: real numpy array of multiplicative weights with the same shape as the data. 
+        bl_len: length of baseline (in 1/[sdf], typically ns)
+        sdf: frequency channel width (typically in GHz)
+        standoff: fixed additional delay beyond the horizon (same units as bl_len)
+        horizon: proportionality constant for bl_len where 1 is the horizon (full light travel time)
+        tol: CLEAN algorithm convergence tolerance (see aipy.deconv.clean)
+        window: window function for filtering applied to the filtered axis. 
+            See aipy.dsp.gen_window for options.        
+        skip_wgt: skips filtering highly flagged rows (unflagged fraction ~< skip_wgt) 
+            Only works properly when all weights are all between 0 and 1.
+        maxiter: Maxinum number of iterations for aipy.deconv.clean to converge.
+
+    Returns:
+        d_mdl: best fit low-pass filter components (CLEAN model) in the frequency domain
+        d_res: best fit high-pass filter components (CLEAN residual) in the frequency domain
+        info: dictionary (1D case) or list of dictionaries (2D case) with CLEAN metadata
+    '''    
+    bl_dly = horizon * bl_len + standoff
+    return high_pass_fourier_filter(data, wgts, bl_dly, sdf, tol=tol,window=window,
+                                    skip_wgt=skip_wgt, maxiter=maxiter)
+
+
+
+
+#import binning
+
+# def delay_filter_aa(aa, data, wgts, i, j, sdf, phs2lst=False, jds=None, 
```

Comment:
Can this commented-out code just be removed?

Reply:
Probably, but it's a useful reminder to anyone looking for old code that these functions used to exist (but I've commented them out since they aren't being unit tested). If in a year no one is clamoring for them, we can delete them.